### PR TITLE
 search/records - no data are loaded at first, then cannot unfold details for individual records #LMR-782

### PR DESCRIPTION
**Pull request by kubedo8**
…ils for individual records #LMR-782

### CODE REVIEW

File: src/app/core/rest/search.service.ts [hunk 1]
```diff
@@ -44,7 +44,11 @@ export class SearchService {
   constructor(private http: HttpClient,
               private store: Store<AppState>,
               private homePageService: HomePageService) {
-    this.store.select(selectWorkspace).subscribe(workspace => this.workspace = workspace);
+    this.store.select(selectWorkspace)
+      .pipe(
+        filter(workspace => !isNullOrUndefined(workspace) && !isNullOrUndefined(workspace.organizationCode) && !isNullOrUndefined(workspace.projectCode))
```
Comment (livthomas) [inline src/app/core/rest/search.service.ts]: I'm just curious. Why do we need to use `isNullOrUndefined` here?

File: src/app/shared/utils/array.utils.ts [hunk 2]
```diff
@@ -50,3 +50,9 @@ export function getArrayDifference<T>(bigArray: T[], smallArray: T[]): T[] {
 
   return bigArray.filter(item => !smallArray.includes(item));
 }
+
+export function arrayIntersection(array1: string[], array2: string[]): string[] {
```
Comment (livthomas) [inline src/app/shared/utils/array.utils.ts]: This method should be made generic in order to be reusable. Look at the method a few lines above.

File: src/app/core/rest/search.service.ts (hunk 1 above)
Comment (kubedo8) [reply src/app/core/rest/search.service.ts]: I wanted to use `workspace && workspace.organizationCode && workspace.projectCode`, but it return string. Do you have any other idea?

File: src/app/shared/utils/array.utils.ts (hunk 2 above)
Comment (kubedo8) [reply src/app/shared/utils/array.utils.ts]: Oh yes :)

File: src/app/core/rest/search.service.ts (hunk 1 above)
Comment (livthomas) [reply src/app/core/rest/search.service.ts]: The typical trick in JavaScript is:
`!!workspace && !!workspace.organizationCode && !!workspace.projectCode`

Comment (justmesr) [reply src/app/core/rest/search.service.ts]: Or `Boolean(workspace && workspace.organizationCode && workspace.projectCode)`, should work the same.

**Pull request merged**